app/src/main/java/com/globant/practice/presentation/presenter/WebClientPresenter.java

```diff
@@ -0,0 +1,79 @@
+package com.globant.practice.presentation.presenter;
+
+import com.globant.practice.presentation.model.WebClientState;
+import com.globant.practice.presentation.view.fragment.WebClientView;
+import javax.inject.Inject;
+
+/**
+ * Manages the possible states of the WebClientFragment using a view model called WebClientState,
+ * load the web page on the WebView and shows o dismiss the progress dialog on the WebClientFragment
+ * Created by jonathan.vargas on 4/05/2017.
+ */
+
+public class WebClientPresenter extends BasePresenter<WebClientView> {
+    private WebClientState webClientState;
+
+    /**
+     * Construct method of the WebClientPresenter and initializes the WebClientState model
+     */
+    @Inject
+    public WebClientPresenter() {
+        webClientState = new WebClientState();
+    }
+
+    /**
+     * Initializes the web page load progress
+     *
+     * @param htmlUrl    address of the web page that will be load
+     * @param detailType indicates if the web page are a profile or a repository
+     */
+    public void loadWebPage(String htmlUrl, String detailType) {
+        webClientState.setHtmlUrl(htmlUrl);
```

Comment:
If this web page is already loading and there is a delay in the connection, you're going to re-load the page again if the screen orientation changes?